core/backend/src/ClassRegistry.ts

```diff
@@ -59,6 +61,21 @@ export class ClassRegistry {
     return schemaClass;
   }
 
+  private static getRootMetaData(iModel: IModelDb, entityMetaData: EntityMetaData): EntityMetaData {
+    let rootClassMetaData = entityMetaData;
+    while (true) {
+      if (rootClassMetaData.baseClasses.length === 0)
```

Comment:
Why isn't this the loop condition instead of `while (true)`?

Reply:
I don't think it looks better:

```ts
let firstBase;
while (rootClassMetaData.baseClasses.length !== 0
  && (firstBase = iModel.getMetaData(rootClassMetaData.baseClasses[0])
  && !firstBase.customAttributes?.some((ca) => ca.ecclass === "IsMixin")) {
  rootClassMetaData = firstBase;
}
```

Reply:
I was only referring to the `baseClasses.length` criterion.

Reply:
there are two condtions that break out of the loop. I think it looks cleaner leaving them as breaks rather than making one of them special as the loop condition